### PR TITLE
feat: implement pool liquidity query in pool manager, deprecate the one in gamm

### PR DESCRIPTION
**Pull request by stackman27**
[Closes: #4802](https://github.com/osmosis-labs/osmosis/issues/4802)

## What is the purpose of the change
Implement total pool liquidity for frontends in pool manager. Deprecate the one in gamm and make it route through pool manager.

## Brief Changelog
-  [#4951](https://github.com/osmosis-labs/osmosis/pull/4951) Implement pool liquidity query in pool manager, deprecate the one in gamm

## Testing and Verifying
tested using localosmosis with state 
output;
```
❯ 
❯ osmosisd q poolmanager total-pool-liquidity 1
liquidity:
- amount: "100000000000"
  denom: stake
- amount: "100000000000"
  denom: uosmo
❯ 
❯ 
❯ osmosisd q poolmanager total-pool-liquidity 2
liquidity:
- amount: "100000000000"
  denom: uion
- amount: "100000000000"
  denom: uosmo
❯ 
❯ 
❯ osmosisd q poolmanager total-pool-liquidity 3
liquidity:
- amount: "100000000000"
  denom: stake
- amount: "10000000000"
  denom: uion
- amount: "200000000000"
  denom: uosmo
❯ 
❯ 
❯ osmosisd q poolmanager total-pool-liquidity 4
liquidity:
- amount: "999999"
  denom: uion
- amount: "19999999999"
  denom: uosmo
```

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? (yes / no)
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? (yes / no)
  - How is the feature or change documented? (not applicable   /   specification (`x/<module>/spec/`)  /  [Osmosis docs repo](https://github.com/osmosis-labs/docs)   /   not documented)

### CODE REVIEW

File: x/poolmanager/router.go [hunk 1]
```diff
@@ -667,3 +667,17 @@ func (k Keeper) createOsmoMultihopExpectedSwapOuts(
 
 	return insExpected, nil
 }
+
```
Comment (mattverse) [inline x/poolmanager/router.go:670]: Can we port over the test casesfor this as well please?

**Review by mattverse — COMMENTED** [review]
LGTM once test cases are ported over

File: x/gamm/client/cli/query.go [hunk 2]
```diff
@@ -34,7 +34,6 @@ func GetQueryCmd() *cobra.Command {
 		GetCmdPoolParams(),
 		GetCmdTotalShares(),
 		GetCmdQueryTotalLiquidity(),
-		GetCmdTotalPoolLiquidity(),
```
Comment (p0mvn) [inline x/gamm/client/cli/query.go:37]: Let's keep the CLI please

**Review by p0mvn — APPROVED** [review]
LGTM.

Please restore x/gamm CLI and Matt's comment about tests

**Comment by stackman27** [comment]
@p0mvn @mattverse imported tests and readded cli!

File: x/concentrated-liquidity/pool_test.go [hunk 3]
```diff
@@ -249,82 +249,6 @@ func (s *KeeperTestSuite) TestCalculateSpotPrice() {
 	s.Require().True(spotPrice.IsNil())
 }
 
-func (s *KeeperTestSuite) TestGetTotalPoolLiquidity() {
```
Comment (p0mvn) [inline x/concentrated-liquidity/pool_test.go:252]: Why is the test removed here? Please add it back

Having a similar test in poolmanager doesn't make this one obsolete

File: x/poolmanager/router_test.go [hunk 4]
```diff
@@ -1990,3 +1990,79 @@ func (suite *KeeperTestSuite) TestSplitRouteExactAmountOut() {
...
+		},
+	}
+
+	for _, tc := range tests {
+		tc := tc
+		s.Run(tc.name, func() {
+			s.SetupTest()
+
+			// Create default CL pool
+			pool := s.PrepareConcentratedPool()
```
Comment (p0mvn) [inline x/poolmanager/router_test.go:2050]: Please also add a test with non-cl pool e.g. balancer

File: x/poolmanager/router_test.go [hunk 5]
```diff
@@ -1990,3 +1990,79 @@ func (suite *KeeperTestSuite) TestSplitRouteExactAmountOut() {
...
+			poolLiquidity:  sdk.NewCoins(defaultPoolCoinTwo, defaultPoolCoinOne, nonPoolCool),
+			expectedResult: defaultCoins,
+		},
+		{
+			// this can happen if someone sends random dust to pool address.
+			name:           "only non-pool coin - does not show up in result",
+			poolId:         1,
+			poolLiquidity:  sdk.NewCoins(nonPoolCool),
+			expectedResult: sdk.Coins(nil),
+		},
```
Comment (p0mvn) [inline x/poolmanager/router_test.go:2036]: Some of these tests are only relevant in the context of concentrated liquidity (looks like this is where they were copied from).

What makes sense to be tested here:
- can route to CL module and get liquidity
- can route to gamm module and get liquidity
- some error cases specific to poolmanager

Please refactor the tests to be relevant for this level of abstraction

**Review by p0mvn — CHANGES_REQUESTED** [review]

File: x/poolmanager/router_test.go (hunk 5 above)
Comment (stackman27) [reply x/poolmanager/router_test.go]: added

File: x/poolmanager/router_test.go (hunk 4 above)
Comment (stackman27) [reply x/poolmanager/router_test.go]: added default asset check. let me know if that suffice

File: x/concentrated-liquidity/pool_test.go (hunk 3 above)
Comment (stackman27) [reply x/concentrated-liquidity/pool_test.go:268]: re-added

**Review by p0mvn — APPROVED** [review]

**Comment by p0mvn** [comment]
Merging since all earlier comments have been addressed

**Pull request merged**